File: codes/rust/121.买卖股票的最佳时机.rs
/*
 * @lc app=leetcode.cn id=121 lang=rust
 *
 * [121] 买卖股票的最佳时机
 */

// @lc code=start
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut min_price = std::i32::MAX;
        let mut max_profit = 0;
        for price in prices {
            min_price = std::cmp::min(min_price, price);
            if price - min_price > max_profit {
                max_profit = price - min_price;
            }
        }
        max_profit
    }
}
// @lc code=end

File: codes/golang/121.买卖股票的最佳时机.go
/*
 * @lc app=leetcode.cn id=121 lang=golang
 *
 * [121] 买卖股票的最佳时机
 */

// @lc code=start
func maxProfit(prices []int) int {
	if len(prices) <= 0 {
		return 0
	}
	res := 0
	min := prices[0]
	for i:=1;i<len(prices);i++ {
		if prices[i] < min {
			min = prices[i]
		}
		if prices[i] - min > res {
			res = prices[i] - min
		} 
	}
	return res
}
// @lc code=end

